### ISSUE TITLE
How to handle "null" JSON fields with custom Deserializer?

### ISSUE THREAD

**Issue by gajicm93** [google/gson#1207]
I want to deserialize JsonNull objects as pure Java null? How do I do that, since the only way I can see right now is to do "instance of" check on the object whether it's JsonNull or not, and then just ignore it if it is?

Is there any smarter way for this? I tried digging through your code to see how you handle it, but didn't find anything much useful, you could also just point me to the part of code where you check for existing but "null" fields.

Thanks.

**Comment by danorton2** [comment]
Can you rephrase your question in the context of this code snippet?
```Java
  JsonObject obj;
  ... // populate obj
  if (obj.has("myfield")) {
    JsonElement e = obj.get("myfield");
    if (e.isJsonNull()) {
      // stuff to do if e is null
    } else {
      // stuff to do if e is not null
    }
  } else {
    // stuff to do if "myfield" is not present
  }
```

**Comment by gajicm93** [comment]
The question is, most simply put, do I have to do "if (e.isJsonNull())", or I can just get null value somehow directly from JsonNull object..

Most specifically, I'd like for e.g. "getAsString()" to return null when invoked on JsonNull object instead of throwing exception.

Can you please tell me where you got this code snippet from? Cause anyway, if this is the way you do it, then there's probably no better way.

**Comment by gajicm93** [comment]
https://github.com/google/gson/blob/6b20e3aaec0b3280f01c7785b5dd391ab6de9cc7/gson/src/main/java/com/google/gson/JsonElement.java#L192

I'd just change the default JsonElement implementation above to return null instead of throwing exception.. That would solve my "problem" completely.. Would it be a breaking change in any way?

**Comment by danorton2** [comment]
Yes, that would be a breaking change, as it changes the functionality and, more generally, the overall concept of the meaning of a null value.

**Comment by danorton2** [comment]
You seem to be asking for this snippet to not throw an exception:
```Java
JsonNull.INSTANCE.getAsString();
```

**Comment by gajicm93** [comment]
@danorton2 Well yeah you could say it that way too, I would like that snippet to just return null. But if it's not possible, I'll just do an instance of check.. 😃

**Comment by danorton2** [comment]
Alternatively, if you get the field from an object, `get()` will return `null` if the field is not present or `JsonNull.INSTANCE` if the field value is null.

**Comment by gajicm93** [comment]
Yeah great thanks, I just wanted to check if there's a better way to do this, but will handle it easily 😃
